src/satosa/backends/saml2.py

```diff
@@ -139,6 +140,13 @@ def authn_request(self, context, entity_id):
                            exc_info=True)
             raise SATOSAAuthenticationError(context.state, "Failed to construct the AuthnRequest") from exc
 
+        if self.sp.config._sp_allow_unsolicited is False:
```

Comment:
Can you let us know if this can be easily addressed in pysaml2 ? I agree with you that this is ugly and we shouldn't let it in